design-documents/storefront/pricing.md

```diff
@@ -0,0 +1,355 @@
+
+## Glossary
+- Customer group - allocate each customer to a group and control store behaviour(catalog restrictions, pricing, discounts, B2B, CMS, payments, shipping, etc) according to which group a customer belongs to  
+- Website, Store View - abstract Magento scopes - https://docs.magento.com/user-guide/configuration/scope.html . Magento modules and third-party extensions may utilize this scope as they want.
+- Base price - the initial product price. Different values could be set per website in Magento monolith.
+- Tier price - have two meanings `customer-specific price` and `volume based price`.  
+- Special price - time based price for the product. Original price is strikedout in the UI. (e.g. was ~~$100.00~~, now $99.00) 
+- Catalog Rule - Magento functionality that allows to apply the same discount to multiple products
+
+
+## Goals
+
+- Support up to 5,000,000 products and 15,000 customer groups
+- Establish efficient sync of prices between Magento monolith and Magento storefront
+- Reduce the size of pricing index
+- Provide reliable support for personalized prices
+
+## Solution
+
+All price dimensions aren't used exclusively for pricing: 
+multiple `websites` may have the same price, but represent different web domains; `customer group` could represent the company in b2b scenarios or class of customer service; `product` may have different pricing on one website only.
+The solution is to reuse data where possible and make separation of dimensions, so `websites` or `customer groups` which are not a part of pricing will not trigger `price index` explosion.
+
+### Price books
+
+The `price book`(price list) is a new entity that holds a list of product's prices for a sub-set of catalog. 
+The `price book` is a non-scoped entity, but it may hold information about linked customers, websites, etc. 
+Instead of direct lookup in price index by customer_group, website and product, we will detect the customer's price book first, then we will extract two price books from the index: default and current `price book`. 
+The resulting product price will be the value from current price book if it's exists, otherwise the product price will be extracted from default price book:
+
+![Price books diagram](pricing/pricebooks.png)
+
+Guidelines:
+* There is no need to resolve price book on each HTTP call. Resolved price book could be stored in JWT during "login" step 
+and reused for consequent requests.
+* In order to minimize "query-time" work, customer should have exactly one resolved price book.
+
+
+### Default price book
+
+A `default price book` is a predefined system price book that contains `base prices` for __ALL__ products in the system. User-defined `price books` may contain only sub-set of products. 
+Default `price book` should be used as fallback storage if the price for a specific product doesn't exist in other resolved pricebook.
```

Comment:
How to determine price for the Default Price book if Product has different prices in several Websites, but product is not assigned to the website marked as "default"?

Reply:
"Default" here is global scope, not some "default website" scope